src/vaults/AsyncRequests.sol

```diff
@@ -198,6 +200,46 @@ contract AsyncRequests is BaseInvestmentManager, IAsyncRequests {
     }
 
     // -- Gateway handlers --
+    /// @inheritdoc IInvestmentManagerGatewayHandler
+    function approvedDeposits(
+        PoolId poolId,
+        ShareClassId scId,
+        AssetId assetId,
+        uint128 assetAmount,
+        D18 pricePoolPerAsset
+    ) external auth {
+        (address asset, uint256 tokenId) = poolManager.idToAsset(assetId);
+
+        balanceSheet.noteDeposit(poolId, scId, asset, tokenId, address(globalEscrow), assetAmount, pricePoolPerAsset);
+        address poolEscrow = address(poolEscrowProvider.escrow(poolId));
+        globalEscrow.authTransferTo(asset, tokenId, poolEscrow, assetAmount);
+    }
+
+    /// @inheritdoc IInvestmentManagerGatewayHandler
+    function issuedShares(PoolId poolId, ShareClassId scId, uint128 shareAmount, D18 pricePoolPerShare) external auth {
+        balanceSheet.issue(poolId, scId, address(globalEscrow), pricePoolPerShare, shareAmount);
+    }
+
+    /// @inheritdoc IInvestmentManagerGatewayHandler
+    function revokedShares(
+        PoolId poolId,
+        ShareClassId scId,
+        AssetId assetId,
+        uint128 assetAmount,
+        uint128 shareAmount,
+        D18 pricePoolPerShare
+    ) external auth {
+        (address asset, uint256 tokenId) = poolManager.idToAsset(assetId);
+        // Lock assets to ensure they are not withdrawn and are available for the redeeming user
+        poolEscrowProvider.escrow(poolId).reserveIncrease(scId, asset, tokenId, assetAmount);
```

Comment:
Since we have a global escrow now again, should we just transfer to global escrow here? Then the reserve logic is not needed at all anymore in the per pool escrow.

Reply:
Ah true, good point!